### PR TITLE
build: aica-package.toml for package-builder:v1

### PR DESCRIPTION
**Pull request by eeberhard**
## Description

- #91

This PR approaches the issue by defining modulo as a collection of packages with package-builder:v1. It doesn't yet make any changes to the CI to build from aica-package.toml.

## Review guidelines

Estimated Time of Review: x minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### Blocked by:
- https://github.com/aica-technology/docker-images/pull/109

(the current metadata for `ros2-ws:v1.0.0-iron` is invalid for package-builder:v1)

### CODE REVIEW

File: CHANGELOG.md
Comment (domire8) [inline CHANGELOG.md:24]: You want to break main now? I thought we'd wait with this

**Review by domire8 — COMMENTED** [review]
> (the current metadata for ros2-ws:v1.0.0-iron is invalid for package-builder:v1)

Why is that?

**Comment by eeberhard** [comment]
> > (the current metadata for ros2-ws:v1.0.0-iron is invalid for package-builder:v1)
> 
> Why is that?

See PR description in https://github.com/aica-technology/docker-images/pull/109:
 
> As part of this followup work, in commit https://github.com/aica-technology/docker-images/commit/4bf3b7b4708c989b05f32e3b331ec5cbe939a07c I also fixed an issue with the image metadata. The tech.aica.image.metadata base.version currently uses BASE_TAG (which is equivalent to the ROS distro, i.e. iron)). I realized that package-builder:v1 will fail to parse the metadata since it expects the base version as a semver string. We will get an error along the lines of:
> 
> ```
> ERROR: failed to solve: invalid AICA metadata "{\"type\":\"base/ws\",\"base\":{\"name\":\"docker.io/library/ros\",\"version\":\"iron\"}}" in image "@aica/foss/ros2-ws": Malformed version: iron
> ```
> In my opinion the simplest and safest approach is to use the VERSION arg instead (which is set from the git tag, i.e. vX.Y.Z-foo). This way the version is parsed as valid, and we can still introspect the ROS distro from it. As far as I am aware, the exact version or formulation of LabelAICABase.Version is not critical as long as we are consistent with it, but please correct me if I am mistaken @LouisBrunner.

File: CHANGELOG.md
Comment (eeberhard) [reply CHANGELOG.md:24]: I didn't plan to release it imminently but I assumed we were gearing up for the breaking release soon. Although it's true that you are staging your other changes on a `breaking` branch currently. In any case we can wait to merge, I just wanted to get it ready here.

Comment (domire8) [reply CHANGELOG.md:24]: Gearing up yes, but just wanted to check if we merge this right away. I see two options:
1. Finish the modulo controllers feature, release that as 4.2 and then start breaking things
2. Tag the current state of main as 4.1.1 and then start breaking things

**Review by LouisBrunner — COMMENTED** [review]
Do we need to change the CI config to use `aica-package.toml`?

**Comment by eeberhard** [comment]
Now that we have v7.4.1 of control libraries (https://github.com/aica-technology/control-libraries/pull/191), this PR can be advanced without breaking modulo. I changed the CI build and test actions to use aica-package.toml instead of the Dockerfile.

If we merge this, I would test the generated `rolling` image for downstream use (i.e. for https://github.com/aica-technology/component-template/pull/1) and if all goes well I would propose deleting the Dockerfile and tagging modulo v4.1.1 to start propagating this metadata.

**Review by domire8 — DISMISSED** [review]
Nice! Would like to see the `feat/mc-interfaces` branch merged and then make a v4.2.0 if possible. Only blocked by #99 now.

**Comment by domire8** [comment]
What should happen with the Dockerfile and the build script now? We're a bit in between atm

**Review by LouisBrunner — APPROVED** [review]

**Comment by eeberhard** [comment]
> What should happen with the Dockerfile and the build script now? 

As I mentioned in the previous comment, if all goes well after merging and testing downstream I think we can delete the Dockerfile and build scripts before / as part of release v4.2

**Review by domire8 — APPROVED** [review]

**Pull request merged**
